id_table.c

```diff
@@ -69,6 +69,8 @@ ITEM_SET_KEY(struct rb_id_table *tbl, int i, id_key_t key)
 static inline int
 round_capa(int capa)
 {
+    /* let power of 2 ints >= 4 not increase capa */
+    if ((capa >= 4) && POW2_P(capa)) return capa;
```

Comment:
would you mind to test https://github.com/ruby/ruby/pull/2086/files as an alternative?